### PR TITLE
feat(module:datapicker): add [nzInputReadOnly] (#4534)

### PR DESCRIPTION
**Pull request by qiurenbo**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number:#4534

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: components/date-picker/picker.component.ts [hunk 1]
```diff
@@ -343,13 +346,13 @@ export class NzPickerComponent implements OnInit, AfterViewInit, OnChanges, OnDe
...
   showClear(): boolean {
-    return !this.disabled && !this.isEmptyValue(this.datePickerService.value) && !!this.allowClear;
+    return !this.inputReadOnly && !this.disabled && !this.isEmptyValue(this.datePickerService.value) && !!this.allowClear;
   }
 
   onClickInputBox(event: MouseEvent, partType?: RangePartType): void {
     event.stopPropagation();
 
-    if (!this.disabled && !this.isOpenHandledByUser()) {
+    if (!this.inputReadOnly && !this.disabled && !this.isOpenHandledByUser()) {
```
Comment (wenqi73) [inline components/date-picker/picker.component.ts:355]: inputReadOnly 只是为了手机端打开时不弹出键盘。打开和 clear 的行为不受影响。
```suggestion
    if (!this.disabled && !this.isOpenHandledByUser()) {
```

File: components/date-picker/picker.component.ts [hunk 2]
```diff
@@ -343,13 +346,13 @@ export class NzPickerComponent implements OnInit, AfterViewInit, OnChanges, OnDe
   }
 
   showClear(): boolean {
-    return !this.disabled && !this.isEmptyValue(this.datePickerService.value) && !!this.allowClear;
+    return !this.inputReadOnly && !this.disabled && !this.isEmptyValue(this.datePickerService.value) && !!this.allowClear;
```
Comment (wenqi73) [inline components/date-picker/picker.component.ts:349]:
```suggestion
    return !this.disabled && !this.isEmptyValue(this.datePickerService.value) && !!this.allowClear;
```

File: components/date-picker/doc/index.en-US.md [hunk 3]
```diff
@@ -45,6 +45,7 @@ The following APIs are shared by nz-date-picker, nz-month-picker, nz-range-picke
 | `[nzAutoFocus]` | get focus when component mounted | `boolean` | `false` | - |
 | `[nzDateRender]` | custom rendering function for date cells (Not support by month-picker/year-picker) | - |`TemplateRef<Date> \| string \| ((d: Date) => TemplateRef<Date> \| string)` | - | - |
 | `[nzDisabled]` | determine whether the nz-date-picker is disabled | `boolean` | `false` | - |
+| `[nzInputReadOnly]` | determine whether the nz-date-picker is read only | `boolean` | `false` | - |
```
Comment (wenqi73) [inline components/date-picker/doc/index.en-US.md:48]: 说明参考下 https://ant.design/components/date-picker/

**Comment by qiurenbo** [comment]
> 感谢 PR！如果能加上测试用例就更好了。

感觉您的建议！我将继续完善

**Pull request closed without merging**